src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/SqlJournal.cs

```diff
@@ -135,8 +135,7 @@ protected override bool ReceivePluginInternal(object message)
         /// <returns>TBD</returns>
         protected override async Task<IImmutableList<Exception>> WriteMessagesAsync(IEnumerable<AtomicWrite> messages)
```

Comment:
so, this is out of left field, but should we consider using `ToImmutableArray` instead of [`ToImmutableList` in PreparePersistentBatch](https://github.com/akkadotnet/akka.net/blob/6ee8b4fd86fc299445dddfd7a2daba3db69b35df/src/core/Akka.Persistence/Journal/WriteJournal.cs#L37-L43) and consider doing so in other places of persistence where we are creating Read-only immutable lists?

Reason being, `ImmutableArray<T>` does implement `IImmutableList<T>`, but uses an array rather than AVL Trees. Might help on locality for PersistAll (and a tiny bit on other things)